File: Task/Linear-congruential-generator/Java/linear-congruential-generator.java
import java.util.stream.IntStream;
import static java.util.stream.IntStream.iterate;

public class LinearCongruentialGenerator {
    final static int mask = (1 << 31) - 1;

    public static void main(String[] args) {
        System.out.println("BSD:");
        randBSD(0).limit(10).forEach(System.out::println);

        System.out.println("\nMS:");
        randMS(0).limit(10).forEach(System.out::println);
    }

    static IntStream randBSD(int seed) {
        return iterate(seed, s -> (s * 1_103_515_245 + 12_345) & mask).skip(1);
    }

    static IntStream randMS(int seed) {
        return iterate(seed, s -> (s * 214_013 + 2_531_011) & mask).skip(1)
                .map(i -> i >> 16);
    }
}

File: Task/Linear-congruential-generator/JavaScript/linear-congruential-generator.js
const mask = (1 << 31) - 1;

function* randBSD(seed) {
    let s = seed;
    while (true) {
        s = (s * 1103515245 + 12345) & mask;
        yield s;
    }
}

function* randMS(seed) {
    let s = seed;
    while (true) {
        s = (s * 214013 + 2531011) & mask;
        yield s >> 16;
    }
}

function take(generator, n) {
    const result = [];
    let count = 0;
    for (const value of generator) {
        if (count >= n) break;
        result.push(value);
        count++;
    }
    return result;
}

// Main execution
console.log("BSD:");
const bsdNumbers = take(randBSD(0), 10);
for (const num of bsdNumbers) {
    console.log(num);
}

console.log("\nMS:");
const msNumbers = take(randMS(0), 10);
for (const num of msNumbers) {
    console.log(num);
}
